WooCommerce/Classes/ViewRelated/ReusableViews/ColumnFlowLayout.swift

```diff
@@ -0,0 +1,44 @@
+import UIKit
+
+/// Renders a grid layout with a specified number of cells per row.
+///
+final class ColumnFlowLayout: UICollectionViewFlowLayout {
+
+    private let cellsPerRow: Int
+
+    init(cellsPerRow: Int,
+         minimumInteritemSpacing: CGFloat = 0,
+         minimumLineSpacing: CGFloat = 0,
+         sectionInset: UIEdgeInsets = .zero) {
+        self.cellsPerRow = cellsPerRow
+        super.init()
+
+        self.minimumInteritemSpacing = minimumInteritemSpacing
+        self.minimumLineSpacing = minimumLineSpacing
+        self.sectionInset = sectionInset
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func prepare() {
+        super.prepare()
+
+        guard let collectionView = collectionView else { return }
+        let marginsAndInsets = sectionInset.left + sectionInset.right
+            + collectionView.safeAreaInsets.left + collectionView.safeAreaInsets.right
+            + minimumInteritemSpacing * CGFloat(cellsPerRow - 1)
+        let itemDimension = ((collectionView.bounds.size.width - marginsAndInsets) / CGFloat(cellsPerRow)).rounded(.down)
+        itemSize = CGSize(width: itemDimension, height: itemDimension)
+    }
+
+    override func invalidationContext(forBoundsChange newBounds: CGRect) -> UICollectionViewLayoutInvalidationContext {
+        guard let context = super.invalidationContext(forBoundsChange: newBounds) as? UICollectionViewFlowLayoutInvalidationContext else {
+            return super.invalidationContext(forBoundsChange: newBounds)
+        }
+        context.invalidateFlowLayoutDelegateMetrics = newBounds.size != collectionView?.bounds.size
```

Comment:
I think that a possible solution for the warning can be to use `invalidationContext.contentSizeAdjustment` inside `invalidationContext` to change the size of the content according to the new bounds.